### ISSUE TITLE
🐛 preferFunctionDeclarations suppression comments being added incorrectly

### ISSUE THREAD

**Issue by yassere** [romefrontend/rome#921]
## Environment

<details>
<summary>$ rome rage --summary</summary>

```
Environment Variables
  Object {
    COLORTERM: "truecolor"
    LANG: "en_US.UTF-8"
    PATH: "/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Users/Yasser/bin:/Users/Yasser/bin"
    TERM: "xterm-256color"
    TERM_PROGRAM: "vscode"
    USER: "Yasser"
  }

User Config
  unset

Rome Version
  0.0.26-dev

Node Version
  14.6.0

Platform
  darwin x64 19.5.0

Terminal Features
  Object {
    background: "unknown"
    colorDepth: 24
    columns: 157
    cursor: true
    hyperlinks: true
    isTTY: true
    unicode: true
  }

Client Flags
  Object {
    clientName: "cli"
    cwd: "/Users/Yasser/Dev/rome"
    realCwd: "/Users/Yasser/Dev/rome"
    silent: false
  }
```
</details>

## Steps To Reproduce

Create a file `foo.ts` with:
```js
let bar;

const foo = () => {};
```
Running `./rome check foo.ts --decisions suppress-lint/js/preferFunctionDeclarations-foo.ts-3` results in:
```js
let bar;
function foo() {} // rome-ignore lint/js/preferFunctionDeclarations
```

With `foo.ts` as:
```js
const foo = () => {};
```
Running `rome check foo.ts --decisions suppress-lint/js/preferFunctionDeclarations-foo.ts-1` results in:
```js
// rome-ignore lint/js/preferFunctionDeclarations
function foo() {} // rome-ignore lint/js/preferFunctionDeclarations
```

## Expected Results

The command should add a single suppression comment above the correct line.

**Comment by yassere** [comment]
The `preferFunctionDeclarations` autofix erases a blank line if there's one before the const function. That seems related to this problem.

**Comment by sebmck** [comment]
I don't think that's related. We retain the newlines by measuring the `loc` lines between two sibling nodes while formatting. I think this is because the suppression logic chooses the last node on a line rather than the first. I've actually rewritten that today to be more efficient so I'll take care of this and make sure the new logic fixes it.

**Comment by sebmck** [comment]
We were never checking for `diagnostic.category === suppression.category` apparently... Fixed in `master`.

**Comment by sebmck** [comment]
Oops meant to close #920.

**Comment by sebmck** [comment]
Fixed by #927.
